module/VuFind/src/VuFind/Controller/AbstractSearch.php

```diff
@@ -356,7 +356,16 @@ protected function getSearchResultsView($setupCallback = null)
             }
 
             foreach ($results->getErrors() as $error) {
-                $this->flashMessenger()->addErrorMessage($error);
+                if (is_array($error)) {
```

Comment:
The flash messenger already supports a specific style of array (see [display code here](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/View/Helper/Root/Flashmessages.php#L104)). Can/should we leverage that instead of having a different intermediate array form here?

Reply:
Oh, nice! Revised the code and added the possibility to add a 'suffix' to the message. This is needed since Blender doesn't have control over the original messages.

Reply:
For the Blender-specific error message a token is great (and now used), but when a backend throws an error, it's not easy to reformat to use tokens while keeping the original message translatable.

Reply:
You're right. I revised it and added a translation string for the combining message so that it can be customized as well. The only complicated part was to get escaping of tokens right so that they're escaped by default but not double-escaped when the actual resulting message is escaped.

Reply:
I've just merged #2359 up through dev-9.0. If you want to merge that in here and resolve conflicts, then I can give this another review later in the week!

Reply:
@demiankatz Merged!